ncm-openstack/src/main/pan/components/openstack/common.pan

```diff
@@ -152,4 +151,22 @@ type openstack_DEFAULTS = {
     Server. NOTE: Nova uses the same config key, but in [neutron] section.
     }
     'metadata_proxy_shared_secret' ? string
-} = dict();
+    @{Driver for security groups}
+    'firewall_driver' ? string = 'neutron.agent.linux.iptables_firewall.IptablesFirewallDriver'
+    @{Use neutron and disable the default firewall setup}
+    'use_neutron' ? boolean = true
+};
+
+
+@documentation {
+    Type to enable RabbitMQ and the message system for OpenStack.
+}
+type openstack_rabbitmq_config = {
+    @{RabbitMQ user to get access to the queue}
+    'user' : string = 'openstack'
+    'password' : string
+    @{Set config/write/read permissions for RabbitMQ service.
+    A regular expression matching resource names for
+    which the user is granted configure permissions}
+    'permissions' : string[3] = list('.*', '.*', '.*')
```

Comment:
This would be much more user friendly as a type, rather than a strange tuple.
Perhaps something like this?
```pan
type openstack_rabbitmq_permissions = {
    'conf' : string = '.*'
    'write' : string = '.*'
    'read' : string = '.*'
}
```

Reply:
Also... why should ncm-openstack be configuring RabbitMQ, rather than a ncm-rabbitmq or metaconfig service?

Reply:
it shouldn't, but the config files are horrible, and you need a component to execute some random commands. so this is way easier, but feel free to open a PR :grin:

Reply:
Heh. :see_no_evil: